### ISSUE TITLE
Question: Spacing and modulation differences between original 1.67?

### ISSUE THREAD

**Issue by stianeklund** [w7sst/MorseRunner#301]
I love that there are up-to-date forks of Morse Runner that aim to improve it, however, I've put off using this variant as it just sounds off / different to me compared to the original Morse Runner 1.67 or Pileup Runner. 

As far as I can tell these are the differences:
1. Morse Runner Community Edition has longer character and letter spacing than 1.67.
2. The monitor level is significantly higher than 1.67; not sure if this is a bug with the original or not.

I've tried to take a look at some of the version history but I couldn't find anything in particular that would change what the modulation sounds like. Perhaps I'm nit picking here but to me the original 1.67 sounds _better_ and more real than 1.84.

In this comparison, both Morse Runner variants are set to use the same parameters:

```
CW Speed: 38
QSK: False
CW Pitch: 450 Hz
RX Bandwidth: 550Hz

Band conditions:
QSB: True
Flutter: True
Activity: 2
```

Here is a screenshot of a comparison of sending `CQ LB1TI TEST`. I've done my best to synchronize the start point in both recordings, I adjusted the volume level of the recording with version 1.84 to try to make it appear more visually similar.

Overall difference:

![Audacity_do7VwAfJcn](https://github.com/w7sst/MorseRunner/assets/4148195/8c009164-ff6b-4e70-8f06-ec33f952e9d9)

Zoomed in:
![Audacity_1dELQandzZ](https://github.com/w7sst/MorseRunner/assets/4148195/ff74208a-91d9-4a09-ad68-5c0dcda9c9d5)

**Comment by w7sst** [comment]
> Morse Runner Community Edition has longer character and letter spacing than 1.67.

You are right, MRCE does have longer character and letter spacing as compared to version 1.67. You are seeing the fix made to Issue #194 where a WPM timing problem and character spacing problem in the original 1.67/1.68 versions was fixed. In that fix, we added some missing spacing between elements of the exchange and also fixed the math involved in generating the CW. These differences are explained below.

### Summary
Version 1.67/1.68 does not have the standard 3U of spacing between characters. It is actually about 2.5U, not 3U. In addition, the letter spacing (inter-character spacing) is about 4.3U, not the standard 7U. This results in a faster WPM than is actually set. In the example presented in this issue, you were using 38 WPM. Version 1.68 was producing correctly formed characters at 38 WPM, but was actually sending the words at 41.8 WPM. When you compare this to version 1.84, you would see a longer inter-character spacing as compared to 1.67. So your ears were correct, 1.84 does have a longer inter-character spacing than 1.67 and is generating correctly-spaced CW.

### Background
The following articles provide insight into the equations used to express the timing used during CW generation.
- https://morsecode.world/international/timing.html
- https://www.arrl.org/files/file/Technology/x9004008.pdf

### Data Collection
1. Select `File | Audio Recording Enabled` menu item before starting the contest.
2. Set speed to 38 WPM, QSK OFF; band conditions to: Flutter ON, QSB ON
2. Click `Run | Single Calls` to a contest
3. Generate a 30 second message (a series of `PARIS` characters). See details in the three tests below.
4. Click `Run` to stop.
5. De-select `File | Audio Recording Enabled` menu item after running it because the state is retained in the .INI file.
6. Rename the `MorseRunner.wav` file for each sample.

### Version 1.84 - WPM Speed Test using F8 debug hook
This first test generate a 30 second sample at 38 WPM by sending 19 occurrences of the word `PARIS`. The generated audio should be 30 seconds in length and character spacing and inter-character spacing should be as expected.

**Steps**
To generate a 30 second sample at 38 WPM...
1. Enter 19 occurrences of `PARIS` by editing your `MorseRunner.ini` file and adding:
```
[Debug]
F8=PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS PARIS
```
2. Follow the steps outlined above to generate the sample. After starting the simulation, press F8 to generate the sample.
3. Rename the `MorseRunner.wav` file to `MR1.84-F8.wav`.
4. Remove the `F8` setting from the `MorseRunner.ini` file.

**Analysis**
1. I opened the sample in an audio tool called Reaper. I think Windows Audacity will provide the same functionality.
2. Insert the first track and attach the input file.
3. Open `File | Project Settings...`
4. Set the `Project BPM` to 1824. This represents 1824 units/minute. This was computed by 48 units/word * 38 words/minute = 1824 units/min.
5. Set the `Time Signature` to 1 / 4.
6. Align the waveform to align within the time signature.

Below is a picture of the sample aligning within the time signature. Notice that all rise/fall edges of each dit or dah align on the unit boundary. There is 1U per dit, 3 units per dah, 1U between each dit/dah, 3U between each character within a word, and 5U between each word (PARIS).
![image](https://github.com/w7sst/MorseRunner/assets/106113737/e86d1936-4980-4615-b4dd-a9ff517ac553)

Also, the measured time elapsed for the full sample was 30 seconds. This shows the generated data was 38 WPM as expected.
![image](https://github.com/w7sst/MorseRunner/assets/106113737/4903c1f6-72ab-42d1-9894-0d8b94ec4e3c)

### Version 1.84 - WPM Speed Test using a series of F4 keystrokes (MyCall)
This next test will prove that we can send a series of `PARIS` characters in succession and achieve the same timing results when compared to using the `F8` debug hook described above. This is done by entering `PARIS` as my callsign and sending it multiple times. Assuming this works, then we can apply the same technique using Version 1.68 in the next step below.

**Steps**
1. Replace your call by entering `PARIS` into the `Station | Call` box.
2. Follow the steps above to generate the sample. Generate 19 occurrences of `PARIS` by pressing `F8` 19 times in succession.
3. Rename the `MorseRunner.wav` file to `MR1.84-paris19.wav.

**Analysis**
1. Add another track and attach the above waveform.
2. Align the sample to the time signature grid
3. Observe that both samples align together. This shows that we can generate a similar sample using successive F4 keystrokes. We will apply this technique with version 1.68.

![image](https://github.com/w7sst/MorseRunner/assets/106113737/9d313b31-cee5-47ed-95b4-cfe3d03ba954)

### Version 1.68 Results - WPM Test using a series of F4 keystrokes (MyCall)
This last test will collect data from MR 1.68.
**Steps**
1. Use MorseRunner version 1.67 or 1.68.
2. Enter `PARIS` into `Station | Call` field (replace your callsign)
3. Follow the steps above to generate the sample. Generate a 30 second sample by pressing `F4` 19 times in succession.
4. Rename the `MorseRunner.ini` file to `MR1.68-paris19.wav`

**Analysis**
1. Add another track and attach the above waveform.
2. Open `Project Settings` and set the `Project BPM` to 1900. Computed as 50 units/word * 38 WPM = 1900 Units/min.
3. Align this sample to the time signature grid.
4. Observe:
  a. intra-character spacing is correct (dit-to-dah spacing) where 1U/dit, 3U/day, 1U between dit/day.
  b. inter-character spacing is ~2.5U, not the expected 3U. This is the bug that was fixed in Issue #194.
  c. inter-word spacing is ~4.3U, not the standard 7U. (Version 1.84 is using 5U inter-letter spacing).
![image](https://github.com/w7sst/MorseRunner/assets/106113737/bb2af4fd-9b6b-43fb-9825-2096c9828579)
*Note that the Time Signature of this third result does not align with the first two samples (tracks).*

Finally, the measured time for the full waveform is ~27.3 seconds. Converting to WPM, we get 19 words/27.3 sec * 60 sec/min = (19*60)/27.3 Words/Min = ~41.8 WPM. This is greater than the expected 38 WPM.
![image](https://github.com/w7sst/MorseRunner/assets/106113737/d0b0b775-32c2-4e0b-b698-3e65b02ea235)

### Conclusions
1. Version 1.68 is generating shortened inter-character and inter-word spacing resulting a faster overall WPM.
2. Version 1.83 (and beyond) is generating correct 3U inter-character spacing and a non-standard 5U of inter-word spacing. The standard is 7U, but chose to use 5U inter-word spacing to better match 1.68's ~4.3. At the time, jumping from ~4.3U to 7U seemed like a lot.
3. Version 1.84 is generating CW that is more correct than 1.67.

I would like hear your thoughts on whether this adequately explains why v1.84 sounds different than 1.67/1.68. Perhaps 1.67 is sounding better because it is a little faster (~43 vs. 38 due to shorted inter-character and inter-letter spacing).

@stianeklund @K6OK @f6fvy @scotthibbs @WR7Q

**Comment by stianeklund** [comment]
@w7sst Thank you for the elaborate reply, your comments explain it more than adequately. I really appreciate you taking the time to give such a well thought out reply.

I do want to emphasize this isn't a big deal for me and I've used MRCE quite a bit since I created this issue. 
At the end of the day as long as it's more correct and aligns better with real-world cw then that's reason enough to use MRCE.
For me increasing the WPM helped a lot to "mitigate" the issue, I can't really notice the differences in inter-char & letter spacing at 40 WPM. Perhaps it's "placebo" that now since I know what the difference is I no longer am bothered by the differences between the versions? 😆

At any rate, thank you for the reply and your efforts with MRCE, I really appreciate it.

73 DE LB1TI / LC9A
